IBX-6709: Page Builder amendments / Drag and drop elements (#261)
ibexa/documentation-user: docs/content_management/create_edit_pages.md

# Create and edit pages

**Page** is a special Content Type.

> You can modify pages with a visual drag-and-drop editor, the Page Builder.

Reviewer: How about adding here a link/anchor link to Page Builder sections?

> Pages are divided into zones into which you can drop various dynamic blocks.

Reviewer: These zones are called drop zones, so maybe it's worth to include it here?

By editing pages, you can customize the layout and content of your website.

Whenever you edit a Page, a [new version](content_versions.md) is created in the repository.

!!! tip
    The Page Content Type contains a **Landing Page** Field Type which manages
    the zones and blocks.
    Any Content item that has the **Landing Page** Field Type behaves like
    a page.

## Create Page

1. In the left panel, go to **Content** -> **Content structure**. Then select a parent Content item and click **Create content**.

> 1. In a slide-out panel, make initial choices in the following fields:

Author: This changes will be covered in separated PR.

    - **Select a language** - from a drop-down list, select the base language for the Content item.
    - **Select a Content Type** - use this field to narrow down the list of Content Type choices displayed below. Then select one of page type, for example, **Landing Page**, and click the **Create** button.
1. In the [Page Builder toolbar](#page-builder-toolbar) click **Fields** and define the page's title and description.

> 1. Click  **Switch layout** and select the layout.

Author: Procedure of "Create Page" will be fixed in separate PR.
For now, in this PR, I restored the master version.

## Edit Page

You can edit Pages using the Page Builder.
To start editing an existing Page, in the Back Office from the Content Tree choose the page that you want you edit and click **Edit** button.

### Page Builder interface

Page Builder has plain and intuitive interface. It is easy to create a Page without having advanced technical skills.

![Page Builder interface](img/page_builder_interface.png)

Page Builder user interface consists of:

A. Drop zone

> B. Elements / Structure view toolbar

Author: As discussed with Justyna - let's use "toolbox".

|![Fields button](page_builder_toolbar_editpreview.png)|Edit main details of the Page, like title and description.|
|![Preview segments](page_builder_toolbar_preview_segment.png)|Access preview of the Page for a given Segment.|
|![Timeline button](page_builder_toolbartimelinetoggler.png)|Access the timeline to preview how the Page changes with time. You can also view the list of all upcoming scheduled events.|
|![View toggler](page_builder_toolbar_devicestoggler.png)|Toggle through to see how the Page is rendered on different devices.|
|![Elements menu](page_builder_toolbarelements.png)|Move Elements / Structure view to the other side of the screen.|
|![Undo](page_builder_toolbar_undo.png)|Undo latest change.|
|![Redo](page_builder_toolbar_redo.png)|Redo latest change.|

Page Builder has two main views that you can use while creating a page:

- Elements toolbar - consists of all available elements that you can use by dragging them and dropping on a drop zone.

![Elements toolbar](img/elements_toolbar.png)

- Structure view - shows a structure of your page, including their division into zones and the blocks that they contain.

![Structure view](img/structure_view.png)

> ### Choose layout

Reviewer: Yes, please mention this button in the table, it appears in the current doc :)

For newly created Page you can choose a [layout](configure_ct_field_settings.md#available-page-layouts) which defines the available zones.

![Switch layout](img/switch_layout_window.png)

Applying a layout divides the Page into the defined zones. The zones are placeholders for Content items.

The page layouts that an editor has access to are up to you to choose.

> In the `Select layouts` section, you can select layouts that you want to be available for the Page.

Reviewer: It's not possible to select multiple layouts, I checked it and it's rather a design issue as there should be radio button I guess. And please change screenshot for selecting layouts, it's updated.

Add a block by dragging it from the menu to an empty place on a zone.
Do not worry about placing blocks in the proper place from the start.
You can reorder blocks in a few ways:

- drag and drop block in the desired location on a drop zone
- access Structure view and use 'Move up' and 'Move down' function in the settings of the block

When you add a new block to the drop zone, drop it in the blue highlighted area. Before you drop it, a bold line appears  - it helps you see the position of the newly added block in relation to other, already added blocks.

![Drop zone line](drop_zone_line.png)

When you add a block by dragging it from Elements menu into the drop zone,

> the block settings panel open immediately where you can configure all block properties.

Author: Toolbox, will be change as discussed.

![Block properties](block_properties.png)

> This is a default behavior. You can globally turn off automatic opening of the block settings panel in the user settings.

Reviewer: Globally as also mentioned in the modal, means globally for all builders.

> This is a default behavior. You can globally turn off automatic opening of the block settings panel in the user settings.

Author: I'd keep the original text from the modal.

First, access your user settings on the right side of the top bar:

![Elements menu](img/user_settings.png "User settings")

Then, go to **My preferences** tab, **Edit** section.
Here, you can find `Automatically open block settings in builder` setting, which, by default, is set up to `enabled` value.

> To change this behavior, click on **Edit** icon, find the setting, and change its value to `disabled`.

Reviewer: As far as I know, first: why, then what to do. So the order here is correct.

![User settings - blocks](img/user_settings_blocks.png "User settings - blocks settings")

[[= product_name =]] comes with a set of ready-to-use Page blocks,
but [developers can add their own]([[= developer_doc =]]/content_management/pages/create_custom_page_block/).

For a list of blocks available out of the box, see [Block reference](block_reference.md).

!!! note

    Before you add a block that involves products, product types, or product categories, make
    sure your that your [user Role](../permission_management/permissions_and_users.md) has
    the `Product/View` and `Product type/View` permission.

    Before you add a block that uses results derived from a [Personalization](../personalization/personalization.md) scenario, 
    for example, [Dynamic targeting](block_reference.md#dynamic-targeting-block) or [Recently added block](block_reference.md#recently-added-block), make sure that the scenario is [properly configured](../personalization/configure_scenarios.md).

You can work with the block, for example, duplicate it, change its position or delete it.

> To access options menu, hover over the block - the toolbar appears.
> Click the icon to see more options.

Reviewer 2: ```suggestion
To access the actions, hover over the block, and the toolbar appears.
Some of them are available after you click the **Options** icon.
```
I've found a name for this icon in figma. It is "options, more, three dots"

Reviewer: To be more precised: it's called kebab :)

Author: "the three dots", as discussed with @juskora.

![Block toolbar](img/block_settings_menu.png "Block toolbar")

> Available options are:

Reviewer 2: ```suggestion
The following actions are available:
```

- Move up block - allows you to change position of the block on the page by moving it up
- Move down block - allows you to change position of the block on the page by moving it down
- Block settings - allows you to access properties window

> - Duplicate - duplicates existing block with its settings by creating its copy

Reviewer 2: ```suggestion
- Duplicate - duplicates existing block with its settings
```

- Refresh - refreshes preview of the block
- Delete - deletes existing block

> For your comfort, you can adjust the size of the block settings window. When you hover over the window frame, blue line appears. 

Reviewer 2: ```suggestion
For your comfort, you can adjust the size of the block settings window. 
When you hover over the window frame, blue line appears. 
```

Also, I feel like this paragrapt belongs in the next section, it could be under the screenshot, right before the "Settings available for blocks are divided into..." paragraph.

After you change the block settings, click **Submit** to save the changes or **Discard** to cancel.

> #### Save page

Reviewer 2: ```suggestion
### Save page
```
I'd put it at the same level as "Add blocks"

There are several options for saving work on the page:

|Option|Description|
|------|-----------|
|Close|Close the page without saving it.|
|Send to review|Save the page and send it to review.|
|Publish / Publish later|[Publish](publish_content.md) the page or schedule publishing for later. |
|Save draft|Save the page draft*.|

> |Delete draft|Delete the page draft.|

Reviewer 2: Publishing the page and deleting the draft actions are not related to saving the work. This is why I was using the convoluted sequence of steps in my original procedure.

Consider going back to the original procedure, renaming it to "Create a new page" and replacing the "Edit the page" step with "Edit page contents". It may start making sense then.

Reviewer: It is related as the article walks a user through point A to point B.
